nucypher/characters/control/serializers.py

```diff
@@ -0,0 +1,118 @@
+
+
+class CharacterControlJsonSerializer(CharacterControlSerializer):
+
+    _serializer = json.dumps
+    _deserializer = json.loads
+
+    def __call__(self, data: Union[bytes, dict], specification: tuple, *args, **kwargs):
+        if isinstance(data, bytes):
+            self.read(request_payload=data, input_specification=specification)
+        elif isinstance(data, dict):
+            self.write(response_data=data, output_specification=specification)
+        else:
+            raise self.SerializerError(f"Invalid serializer input types: Got {data.__class__.__name__}")
+
+    @staticmethod
+    def __build_response(response_data: dict):
+        response_data = {'result': response_data, 'version': str(nucypher.__version__)}
+        return response_data
+
+    @staticmethod
+    def __validate_input(request_data: dict, input_specification: tuple) -> bool:
+        for field in input_specification:
+            if field not in request_data:
+                raise CharacterControlSpecification.MissingField(f"Request is missing the '{field}' field")
+        return True
+
+    @staticmethod
+    def __validate_output(response_data: dict, output_specification: tuple) -> bool:
+        for field in output_specification:
+            if field not in response_data['result']:
+                raise CharacterControlSpecification.InvalidResponseField(f"Response is missing the '{field}' field")
+        return True
+
+    def read(self, request_payload: bytes, input_specification: tuple) -> dict:
+        if not request_payload:
+            return dict()  # Handle Empty Request Body
+        try:
+            request_data = CharacterControlJsonSerializer._deserializer(request_payload)
+
+        except JSONDecodeError:
+            raise self.SerializerError(f"Invalid protocol input: got {request_payload}")
+
+        self.__validate_input(request_data=request_data, input_specification=input_specification)
+        return request_data
+
+    def write(self, response_data: dict, output_specification) -> bytes:
+        response_data = self.__build_response(response_data=response_data)
+        response_payload = CharacterControlJsonSerializer._serializer(response_data)
+        self.__validate_output(response_data=response_data, output_specification=output_specification)
+        return response_payload
+
+
+class AliceCharacterControlJsonSerializer(CharacterControlJsonSerializer):
+
+    @staticmethod
+    def load_create_policy_input(request: dict):
+        parsed_input = dict(bob_encrypting_key=bytes.fromhex(request['bob_encrypting_key']),
+                            bob_verifying_key=bytes.fromhex(request['bob_verifying_key']),
+                            label=b64decode(request['label']),
+                            m=request['m'],
+                            n=request['n'])
+        return parsed_input
+
+    @staticmethod
+    def dump_create_policy_output(response):
+        unicode_label = response['label'].decode()
```

Comment:
It does not have to be as deep as pyUmbral since there may be advantages of using bytes labels in a public API, However, there is some direct convenience added to nucypher when handling Unicode labels in URLs.